### PR TITLE
Raw RMC support

### PR DESCRIPTION
**Pull request by jonbarrow**
~~Draft for now because we have to update `titles.json`~~

Relies on https://github.com/PretendoNetwork/HokakuCTR/pull/6

PR implements support for raw RMC pcaps made by [HokakuCTR](https://github.com/PretendoNetwork/HokakuCTR). There's a couple hacks implemented to get this working, but I think they're okay and are done so we don't have to refactor a massive portion of the parser/renderer

Also swapped to using JSDoc imports for types. Before, to use a type in JSDoc we were importing the real script with `require`, which could create circular dependencies (ran into this during initial testing of this implementation). Now we just import the type with JSDoc. We had to disable ESLint for those lines anyway which was not great

For those reviewing, *everything* changed inside `src/protocols` is just the JSDoc changes. Those files can be safely ignored

Tested using Mario Kart 7

Screenshot from 2023-10-11 03-54-08

### CODE REVIEW

**Comment by jonbarrow** [comment]
Looks like this would also fix #17. A bug was introduced before on [this line](https://github.com/PretendoNetwork/nex-viewer/blob/465b72a456db80e00bb0948cbe6e1bca6252b613/src/connection.js#L287C5-L287C5) which makes requests bodies not get handled correctly (since a request will always fail `isSuccess()`, since `responseStatus` is not present in requests)

I added a fix for this in this PR, and I suspect that's what's causing the linked issue

**Comment by jonbarrow** [comment]
Marking this as ready for review now that `titles.json` has been updated. We can't merge this until the HokakuCTR PR is merged, but it's ready for review

**Comment by DaniElectra** [comment]
> Looks like this would also fix #17. A bug was introduced before on [this line](https://github.com/PretendoNetwork/nex-viewer/blob/465b72a456db80e00bb0948cbe6e1bca6252b613/src/connection.js#L287C5-L287C5) which makes requests bodies not get handled correctly (since a request will always fail `isSuccess()`, since `responseStatus` is not present in requests)
> 
> I added a fix for this in this PR, and I suspect that's what's causing the linked issue

I'm almost certain that's a different issue, related with the fact that we don't verify if the login response is successful or not https://github.com/PretendoNetwork/nex-viewer/blob/465b72a456db80e00bb0948cbe6e1bca6252b613/src/connection.js#L317-L319

**Comment by jonbarrow** [comment]
> I'm almost certain that's a different issue, related with the fact that we don't verify if the login response is successful or not

Upon further inspection yes you're right. I didn't notice this either. I had ran into the bug I mentioned, and seeing as how it would only be caused when `packet.rmcData.body` is `undefined` (which also happens to requests due to the above mentioned bug), I assumed that was the cause

But looking at the screenshot again, it already made it past the request and the error is thrown on the response

File: src/parser.js
Comment (jonbarrow) [reply src/parser.js:271]: Correct. Lot's of 2am programming here

**Comment by jonbarrow** [comment]
HokakuCTR has been updated to support this and the required changes have been made here. This is officially ready for review

**Comment by jonbarrow** [comment]
@DaniElectra any objections to this?

File: src/connection.js
Comment (DaniElectra) [inline src/connection.js:378]: We are using `this.accessKey` when checking for game-specific protocols

```suggestion
					this.title = title;
					this.accessKey = title.access_key;
```

**Review by DaniElectra — COMMENTED** [review]
Looks good, but we are also waiting on #18 to be merged. We can merge and fix the issue or let Trace fix it.

**Comment by jonbarrow** [comment]
@DaniElectra should be good now?

File: titleids.txt
Comment (DaniElectra) [inline titleids.txt:16]: I guess this is redundant now

Comment (jonbarrow) [reply titleids.txt]: Removed it

**Review by DaniElectra — APPROVED** [review]
Looks good to me

**Pull request merged**
